pages/transaction-detail.vue

```diff
@@ -126,13 +232,57 @@ export default {
       }
       const result = (await connection.api.Transaction.getTransactions(query)).transactions;
       this.transaction = result[0]
+      console.log(this.transaction);
     } catch (error) {
       console.log(error.message);
       // error({ statusCode: 404, message: 'Oops...' });
     }
 
     this.date = moment(slots.getRealTime(this.transaction.timestamp)).format('YYYY-MM-DD hh:mm:ss');
 
+    this.args = JSON.parse(this.transaction.args);
+
+    switch (this.transaction.type) {
+      case 0:
+        this.args[0] = new BigNumber(this.args[0]).dividedBy(1e8).toFixed();
+        break;
+      case 1:
+        this.username = this.args[0];
+        break;
+      case 2:
+        this.secondPublicKey = this.args[0];
+        break;
+      case 3:
+        this.lockHeight = new BigNumber(this.args[0]).toFixed();
+        this.amount = new BigNumber(this.args[1]).dividedBy(1e8).toFixed();
+        break;
+      case 4:
+      case 5:
+        this.voteList = this.args[0];
+        break;
+      case 100:
+        this.username = this.args[0];
+        this.desc = this.args[1];
+        break;
+      case 101:
+        this.username = this.args[0];
+        this.desc = JSON.stringify(this.args[1]);
+        this.maximum = new BigNumber(this.args[2]).dividedBy(1e8).toFixed();
+        this.precision = this.args[3];
+        break;
+      case 102:
+        this.currency = this.args[0];
+        this.amount = new BigNumber(this.args[1]).dividedBy(1e8).toFixed();
+        break;
+      case 103:
+        this.currency = this.args[0];
+        this.amount = new BigNumber(this.args[1]).dividedBy(1e8).toFixed();
```

Comment:
@xpgeng here you need to fetch the `precision` of the asset first. Because not for all assets the precision is `8`. You then need to divide by the appropriate precision.

Reply:
there is some code in the gny-wallet that is doing this. Have a look